test/stdlib/FileTest_test.rb

```diff
@@ -0,0 +1,188 @@
+require_relative "test_helper"
+
+class FileTestSingletonTest < Test::Unit::TestCase
+  include TypeAssertions
+
+  testing "singleton(::FileTest)"
+
+  def test_blockdev?
+    assert_send_type  "(::String file_name) -> bool",
+                      FileTest, :blockdev?, __FILE__
+    assert_send_type  "(::IO file_name) -> bool",
+                      FileTest, :blockdev?, io_open
+  end
+
+  def test_chardev?
+    assert_send_type  "(::String file_name) -> bool",
+                      FileTest, :chardev?, __FILE__
+    assert_send_type  "(::IO file_name) -> bool",
+                      FileTest, :chardev?, io_open
+  end
+
+  def test_directory?
+    assert_send_type  "(::String file_name) -> bool",
+                      FileTest, :directory?, __FILE__
+    assert_send_type  "(::IO file_name) -> bool",
+                      FileTest, :directory?, io_open
+  end
+
+  def test_empty?
+    assert_send_type  "(::String file_name) -> bool",
+                      FileTest, :empty?, __FILE__
+    assert_send_type  "(::IO file_name) -> bool",
+                      FileTest, :empty?, io_open
+  end
+
+  def test_executable?
+    assert_send_type  "(::String file_name) -> bool",
+                      FileTest, :executable?, __FILE__
+  end
+
+  def test_executable_real?
+    assert_send_type  "(::String file_name) -> bool",
+                      FileTest, :executable_real?, __FILE__
+  end
+
+  def test_exist?
+    assert_send_type  "(::String file_name) -> bool",
+                      FileTest, :exist?, __FILE__
+    assert_send_type  "(::IO file_name) -> bool",
+                      FileTest, :exist?, io_open
+  end
+
+  def test_exists?
```

Comment:
`exists?` was [removed in 3.0](https://github.com/ruby/ruby/commit/bf97415c02b11a8949f715431aca9eeb6311add2) (?). Deleting this test case would pass the test.